eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1648,6 +1648,11 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
     case Event(WatchFundingSpentTriggered(tx), d: PersistentChannelData) if tx.txid == d.commitments.localCommit.commitTxAndRemoteSig.commitTx.tx.txid =>
       log.warning(s"processing local commit spent in catch-all handler")
       spendLocalCurrent(d)
+
+    // forward unknown messages that originate from loaded plugins
+    case Event(unknownMsg: UnknownMessage, _) if nodeParams.pluginMessageTags.contains(unknownMsg.tag) =>
```

Comment:
This is probably worth a unit test to make sure we don't inadvertently break that forwarding code in a future refactoring.

Reply:
I don't think it's worth creating a new test spec file, but I don't think `ChannelDataSpec` is the right place for it either. This is something that happens during channel's lifetime, so I guess `NormalStateSpec` is the most ok place to put it, above the `recv CurrentBlockCount` tests.

But it makes me wonder, do you really need that handler inside the `Channel` actor? This is really something that should directly go to the `Peer` actor, can't the `PeerSwap` plugin obtain a reference to the peer instead of going through the channel? It would be much cleaner.

Reply:
> But it makes me wonder, do you really need that handler inside the `Channel` actor? This is really something that should directly go to the `Peer` actor, can't the `PeerSwap` plugin obtain a reference to the peer instead of going through the channel? It would be much cleaner.

One option would be to initialize a swap actor with the remote peer's `NodeId` and a reference to the `switchboard`. Then we could do something like:

```scala 
switchboard ! ForwardUnknownMessage(peerNodeId, unknownMessage)
```

and add to `Switchboard`:
```scala
case class Switchboard.ForwardUnknownMessage(nodeId: PublicKey, msg: UnknownMessage)

    case ForwardUnknownMessage(remoteNodeId, msg) =>
      getPeer(remoteNodeId) match {
        case Some(peer) => peer ! Peer.OutgoingMessage(msg, activeConnection)
      }
```
But only the channel keeps track of the current `activeConnection` so we could perhaps change `OutgoingMessage` in  `Peer` to make the `peerConnection` parameter optional. 

```scala
  case class OutgoingMessage(msg: LightningMessage, peerConnection: Option[ActorRef])
```

 ¯\\_(ツ)_/¯

Reply:
I see, the issue is that we currently don't have any places in the codebase where we send outgoing unknown messages yet, I thought we had that in the `Peer` but we actually only have code to _receive_ unknown messages.

The question is then what guarantees do we want to provide for outgoing unknown messages? The reason the channel keeps track of the active connection is because for channel messages, we want to avoid sending them when a reconnection happened because we have an explicit reconnection protocol to reconcile messages that weren't acked. I don't think we want that feature for unknown messages, do we? We simply always want them to be sent on the active connection, right?

If that's correct, we just need the `Peer` to define a new command `RelayUnknownMessage`, similar to `RelayOnionMessage`. We can then also have the `switchboard` provide a way to send it to a `remoteNodeId` with the `ForwardUnknownMessage` you suggest.

Reply:
> I don't think we want that feature for unknown messages, do we? We simply always want them to be sent on the active connection, right?

I agree that there's no assumed re-connection protocol  for sending `UnknownMessages`. I'll make these changes to my Peerswap plugin branch to double check that it is sufficient to send a `RelayUnknownMessage` messages to the`Peer` via the `switchboard` instead of using a `Register.ForwardShortId` message.

Reply:
Changed to forward unknown messages via the switchboard in [0532835](https://github.com/ACINQ/eclair/pull/2495/commits/0532835094c1eeb277b54b4657d754d7db3075e2).

I also refactored the `peerswap-plugin` branch to test that this change is sufficient for that use case.

Reply:
PR #2465 adds `Register.ForwardNodeId` which does something similar to the the new `Switchboard.ForwardUnknownMessage` proposed by this PR. Using `Register.ForwardNodeId` does not add redundant code to `Switchboard` and includes `replyTo` with a failure message if the peer does not exist.

Once PR #2465 is merged I'll update this PR to use it.

Reply:
[Rolled back (510b695)](https://github.com/ACINQ/eclair/pull/2495/commits/510b695f5a6aea2c90255d5bc54d2607a601b25c) addition of `Switchboard.ForwardUnknownMessage`. Instead of using the `switchboard` to forward to unknown messages to a remote peer,  plugins should query the `switchboard` using `GetPeerInfo` to retrieve the actor ref of the remote peer and use the new `Peer.RelayUnknownMessage` event to forward unknown messages.